NodeJS/__unit__/AgentClientMockito.test.ts

```diff
@@ -0,0 +1,42 @@
+import { AgentClient } from "../src/AgentClient";
+import { RestClient } from "../src/RestClient";
+import { mock, when, anyString, reset, instance, verify } from "ts-mockito";
+
+class AgentClientProtected extends AgentClient{
+    public fetchDefaultSkillNumber(){
+        return super.fetchDefaultSkillNumber()
+    }
+}
+
+describe("AgentClient.ts tests with mocked RestClient", () => {
+    let agentClient: AgentClientProtected;
+    let restClientMock: RestClient = mock(RestClient)
+    let restClientInstance: RestClient
+
+    beforeEach(() => {
+        reset(restClientMock)
+        restClientInstance = instance(restClientMock)
+
+    });
+    test("getDefaultSkillId should return default skill id", async () => {
+        let data = { "skillResponses": { '1': [{ name: 'DEFAULT_SKILL', number: 100 }] } }
```

Comment:
nit: `skillResponses` does not need to be wrapped in quotes

Reply:
corrected.